handlers/handlers.go

```diff
@@ -246,11 +249,83 @@ func HandleNoteApiRequest(
 
 		fmt.Fprint(responseWriter, string(notesInJson))
 
+	case http.MethodPost:
+		type NoteForm struct {
+			Content string `json:"content"`
+		}
+
+		noteForm := new(NoteForm)
+
+		if err := json.NewDecoder(request.Body).Decode(noteForm); err != nil {
+			http.Error(responseWriter, err.Error(), http.StatusBadRequest)
+			return
+		}
+
+		if len(strings.TrimSpace(noteForm.Content)) == 0 {
+			http.Error(responseWriter, "Note content cannot be empty or just whitespace", http.StatusBadRequest)
+			return
+		}
+
+		note := &models.Note{
+			AuthorId:     models.UserId(userId),
+			Content:      noteForm.Content,
+			CreationTime: time.Now().UTC(),
+		}
+
+		noteId, err := noteservice.StoreNewNote(note)
+		if err != nil {
+			http.Error(responseWriter, err.Error(), http.StatusInternalServerError)
+			return
+		}
+
+		responseWriter.Write([]byte(`{NoteId: "` + strconv.FormatInt(int64(noteId), 10) + `"}`))
+		responseWriter.WriteHeader(http.StatusCreated)
+
 	default:
-		respondWithMethodNotAllowed(responseWriter, http.MethodGet)
+		respondWithMethodNotAllowed(responseWriter, http.MethodGet, http.MethodPost)
 	}
 }
 
+func HandleCategoryApiRequest(
+	responseWriter http.ResponseWriter,
+	request *http.Request,
+	userId models.UserId,
+) {
+	switch request.Method {
+	case http.MethodPost:
```

Comment:
we should come to some agreement on this. The main issue with it being a post is that i would expect it to return some id. And since we're specifying an id in the request this feels much more like a PUT. Though that means the `noteId` needs to move into the url. Whether in the query params or as part of the url I don't have a strong preference. We should just be consistent in whatever we choose.

Reply:
I'm okay with removing the distinction between post and put for note category. I agree right now the distinction doesn't make sense. So this handler will eventually have a `get` a `put` and a `delete` but no `post` ya?

Reply:
sounds right